### PR TITLE
Set more ANGLE shader translation properties for WebGL 2

### LINKED ISSUE #25034: WebGL2 program link error when using uniform blocks
The following code has been minimized from the WebGL conformance suite's uniform buffer test (tests/wpt/webgl/tests/conformance2/buffers/uniform-buffers.html).

https://pastebin.com/4xQysf6A

When running it with Servo (eg. `./mach run test.html --pref dom.webgl2.enabled --dev --headless`) it manages to compile the shaders, but unable to link them together into a program. The following log output is produced:

```
shader 'vshader' compiled? true
shader log for 'vshader': 
shader 'fshadernamed' compiled? true
shader log for 'fshadernamed': 
program linked? false
program log: error: linking with uncompiled/unspecialized shader
DONE
```

(ie. **error: linking with uncompiled/unspecialized shader**).
The code does work in Firefox (69) and Chromium (78), and produces the expected result:

```
shader 'vshader' compiled? true
shader log for 'vshader': 
shader 'fshadernamed' compiled? true
shader log for 'fshadernamed': 
program linked? true
program log: 
DONE
```

(Tested on Ubuntu 18.04, in both headless and non-headless mode)

cc @jdm @zakorgy

### PR DESCRIPTION
**Pull request by jdm**
Fixes #25034.

### CODE REVIEW

**Review by asajeffrey — APPROVED** [review]
Some nits but mostly LGTM.

File: components/script/dom/webglshader.rs [hunk 1]
```diff
@@ -81,6 +81,121 @@ impl WebGLShader {
     }
 }
 
+// Based on https://searchfox.org/mozilla-central/rev/efdf9bb55789ea782ae3a431bda6be74a87b041e/gfx/angle/checkout/src/compiler/translator/ShaderLang.cpp#173
+fn default_validator() -> BuiltInResources {
+    BuiltInResources {
+        // Constants.
```
[the remark this answers is not in the document]
Comment (jdm) [reply components/script/dom/webglshader.rs]: You mean the comment that is three lines above the line you left this review comment on?

**Comment by jdm** [comment]
```
  ▶ CRASH [expected ERROR] /_webgl/conformance2/uniforms/incompatible-texture-type-for-sampler.html
  │ 
  │ 
  │ 
  │ 
  │ [2019-12-19T22:21:01Z ERROR canvas::webgl_thread] Last GL operation failed: TexImage2D { target: 3553, level: 0, effective_internal_format: 6406, size: 1×1, format: Alpha, data_type: UnsignedByte, effective_data_type: 5121, unpacking_alignment: 4, alpha_treatment: None, y_axis_treatment: AsIs, pixel_format: None, data: IpcSharedMemory { os... }
  │ Unexpected WebGL error: 0x501 (1281) [TexImage2D { target: 3553, level: 0, effective_internal_format: 6406, size: 1×1, format: Alpha, data_type: UnsignedByte, effective_data_type: 5121, unpacking_alignment: 4, alpha_treatment: None, y_axis_treatment: AsIs, pixel_format: None, data: IpcSharedMemory { os... }] (thread WebGL thread, at components/canvas/webgl_thread.rs:1819)
 
  ▶ ERROR [expected OK] /_webgl/conformance2/programs/gl-get-frag-data-location.html
  │   → gl.getFragDataLocation is not a function
  │ 
  │ runTests@http://web-platform.test:8000/_webgl/conformance2/programs/gl-get-frag-data-location.html:95:17
  └ @http://web-platform.test:8000/_webgl/conformance2/programs/gl-get-frag-data-location.html:83:3
  ▶ Unexpected subtest result in /_webgl/conformance2/programs/gl-get-frag-data-location.html:
  │ FAIL [expected PASS] WebGL test #2: successfullyParsed should be true (of type boolean). Was undefined (of type undefined).
  │   → assert_true: successfullyParsed should be true (of type boolean). Was undefined (of type undefined). expected true got false
  │ 
  │ reportTestResultsToHarness/<@http://web-platform.test:8000/_webgl/js/js-test-pre.js:123:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1984:25
  │ test@http://web-platform.test:8000/resources/testharness.js:548:30
  │ reportTestResultsToHarness@http://web-platform.test:8000/_webgl/js/js-test-pre.js:122:7
  │ testFailed@http://web-platform.test:8000/_webgl/js/js-test-pre.js:249:5
  │ shouldBe@http://web-platform.test:8000/_webgl/js/js-test-pre.js:406:9
  │ shouldBeTrue@http://web-platform.test:8000/_webgl/js/js-test-pre.js:432:29
  └ @http://web-platform.test:8000/_webgl/js/js-test-post.js:24:1
```

**Comment by jdm** [comment]
The gl-get-frag-data-location.html is new because the test previously returned early when the compiling the program failed. Now that it doesn't, we hit different errors.

The new GL error is almost certainly caused by disabling the compat mode, since it's using ALPHA.

**Comment by asajeffrey** [comment]
Oh good, compat mode again.

**Comment by jdm** [comment]
@bors-servo retry
* #25459

**Pull request merged**